openstudiocore/src/model/SurfacePropertyConvectionCoefficients.cpp

```diff
@@ -0,0 +1,399 @@
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **********************************************************************************************************************/
+
+#include "SurfacePropertyConvectionCoefficients.hpp"
+#include "SurfacePropertyConvectionCoefficients_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+#include "Surface.hpp"
+#include "Surface_Impl.hpp"
+#include "SubSurface.hpp"
+#include "SubSurface_Impl.hpp"
+#include "InternalMass.hpp"
+#include "InternalMass_Impl.hpp"
+#include "Schedule.hpp"
+#include "Schedule_Impl.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_SurfaceProperty_ConvectionCoefficients_FieldEnums.hxx>
+
+#include "../utilities/units/Unit.hpp"
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const IdfObject& idfObject,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == SurfacePropertyConvectionCoefficients::iddObjectType());
+  }
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == SurfacePropertyConvectionCoefficients::iddObjectType());
+  }
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const SurfacePropertyConvectionCoefficients_Impl& other,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& SurfacePropertyConvectionCoefficients_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    return result;
+  }
+
+  IddObjectType SurfacePropertyConvectionCoefficients_Impl::iddObjectType() const {
+    return SurfacePropertyConvectionCoefficients::iddObjectType();
+  }
+
+  bool SurfacePropertyConvectionCoefficients_Impl::setSurface(const ModelObject &surface) {
+    return setPointer(OS_SurfaceProperty_ConvectionCoefficientsFields::SurfaceName, surface.handle());
```

Comment:
A couple of things:

 - I can't add that method to `PlanarSurface` because it needs to be on `Surface`, `SubSurface`, and `IntMass`. Not all of those inherit from `PlanarSurface`. I'm planning on just putting in a separate getter on each of those.
 - So, when I `setSurface` on the `SurfacePropertyConfectionCoefficients`, if I have it remove the existing surface coefficients, should I just delete them? Otherwise, they'll be left without a surface to point to, which I specifically designed this API to never allow. 
 - Same concern with blanking out the surface field on a clone, then we'll have one of these objects without a surface, which I was trying to avoid. Am I missing something?